### PR TITLE
Course length fees

### PR DESCRIPTION
**Pull request by defong**
### Context
A user can enrich a non-salaried course's length and fees 
A user can view the course summary
 
### Changes proposed in this pull request
A user can enrich a non-salaried course's length and fees 
A user can view the course summary

### Guidance to review
Course summary
Non-salaried course
summary-with-course-length-and-fee
Salaried course
summary-without-course-length-and-fee

Course length and fees edit page
Edit
course-length-and-fee
Validation
course-length-and-fee-validation

### CODE REVIEW

**Comment by defong** [comment]
['Course length and fees' enrichment and saving [non-salaried]](https://trello.com/c/EfPsXztQ/130-course-length-and-fees-enrichment-and-saving-non-salaried)

File: src/ui/Controllers/CourseController.cs [hunk 1]
```diff
@@ -278,7 +336,8 @@ private VariantViewModel LoadViewModel(UserOrganisation org, Course course, bool
                 LiveSearchUrl = searchAndCompareUrlService.GetCoursePageUri(org.UcasCode, courseVariant.ProgrammeCode),
                 AllowPreview = featureFlags.ShowCoursePreview,
                 AllowPublish = featureFlags.ShowCoursePublish,
-                AllowLiveView = featureFlags.ShowCourseLiveView
+                AllowLiveView = featureFlags.ShowCourseLiveView,
+                IsSalary = course.ProgramType.Equals("SS", StringComparison.InvariantCultureIgnoreCase)
```
Comment (timabell) [inline src/ui/Controllers/CourseController.cs]: This is the wrong place for this logic, though I think we might be worrying about that later.

File: src/ui/ViewModels/FromUcasViewModel.cs [hunk 2]
```diff
@@ -17,6 +17,7 @@ public class VariantViewModel
         public CourseEnrichmentViewModel CourseEnrichment { get; set; }
         public CourseRouteDataViewModel RouteData { get; set; }
         public Uri LiveSearchUrl { get; set; }
+        public bool IsSalary { get; set; }
```
Comment (timabell) [inline src/ui/ViewModels/FromUcasViewModel.cs]: `IsSalaried` would be better english I think

minor

File: src/ui/Views/Course/Fees.cshtml [hunk 3]
```diff
@@ -0,0 +1,115 @@
+@model CourseFeesEnrichmentViewModel
+@{
+  Layout = "_Layout";
+  var course_length = ViewData.ModelState["CourseLength"] != null && ViewData.ModelState["CourseLength"].Errors.Any();
+  var fee_uk_eu = ViewData.ModelState["FeeUkEu"] != null && ViewData.ModelState["FeeUkEu"].Errors.Any();
+  var fee_iternational = ViewData.ModelState["FeeInternational"] != null && ViewData.ModelState["FeeInternational"].Errors.Any();
+  var fee_details = ViewData.ModelState["FeeDetails"] != null && ViewData.ModelState["FeeDetails"].Errors.Any();
+  var financial_support = ViewData.ModelState["FinancialSupport"] != null && ViewData.ModelState["FinancialSupport"].Errors.Any();
```
Comment (timabell) [inline src/ui/Views/Course/Fees.cshtml]: I suggest renaming these to `has_financial_support_errors` etc

minor

File: src/ui/Controllers/CourseController.cs (hunk 1 above)
Comment (defong) [reply src/ui/Controllers/CourseController.cs]: Agreed wrong place as there is limited right place for it

File: src/ui/Views/Course/Fees.cshtml [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        </div>
+         <div id="InternationalCourseFees-wrapper" class="govuk-form-group @(fee_iternational ? " govuk-form-group--error " : " ")">
+          <label asp-for="FeeInternational" class="govuk-label">Fee for international students (optional)</label>
+          @if (fee_iternational) {
+            <span asp-validation-for="FeeInternational" class="govuk-error-message"></span>
+          }
+          <div class="govuk-currency-input">
+            <div class="govuk-currency-input__inner">
+              <span class="govuk-currency-input__inner__unit">£</span>
+              <input asp-for="FeeInternational" type="number" class="govuk-currency-input__inner__input govuk-input govuk-input--width-10"></textarea>
```
Comment (timabell) [inline src/ui/Views/Course/Fees.cshtml]: input / textarea tags mismatched @dankmitchell 

reject

File: src/ui/ViewModels/FromUcasViewModel.cs (hunk 2 above)
Comment (defong) [reply src/ui/ViewModels/FromUcasViewModel.cs]: That bool will change all together

**Pull request merged**